Feature/134209527/redesign authentication ui

andela-uenabulele opened the pull request:
**What does this PR do?**

It redesigns the login/signup page using material design specifications.

**Description of Task that was completed?**

Redesigned signup.html and signin.html and added css styling.
Re-integrated the already existing angular module into the HTML
How should this be manually tested?

Run this app locally or click here to visit our Heroku review app.
Click on Login or signup at the top right corner of the page.
Any background context you want to provide?

**What are the relevant pivotal tracker stories?**

#134209527 Re-Design UI for user authentication screen

Screenshots (if appropriate)
![screen shot 2016-12-14 at 11 13 39 am](https://cloud.githubusercontent.com/assets/23451840/21178152/80c2b398-c1ee-11e6-8bdf-27813fd7fd0f.png)


![screen shot 2016-12-14 at 11 15 04 am](https://cloud.githubusercontent.com/assets/23451840/21178174/9c554922-c1ee-11e6-888e-b7d8160c929d.png)

Questions:
None

Commit:
feature(fan-base): add facebook-like button to landing page [WIP #134678595]

Commit:
feature(style-guide): refactor index.html for readability [WIP #134678595]

Commit:
bug(fan-base): fix invisible social media share buttons [WIP #134678595]

Commit:
bug(UX): fix links not working on landing page [Finishes #134678595]

Commit:
refactor(style-guide): fix issues with hound violations [Fixes #134678595]

Commit:
feature(ui-ux): implement login ui design [Start #134209527]

Commit:
feature(ui-ux): implement signup ui design [WIP #134209527]

Commit:
chore(style-guide): refactor code to meet style guide specification [Finishes #134209527]

Commit:
chore(refactor) fix houndCI issues [Fixes 134209527]

Commit:
feature(routes): configure routes for play-with page [Finishes #134209527]

this commit also adds some css properties to the play-with page

Commit:
feature(user-interface): add css to 'play-with' page for mobile responsiveness [Finishes #134209527]

Commit:
refactor(style-guide): refactor code to fix houndCI issues [Finishes #134209527]

Commit:
refactor(style-guide): refactor code to fix houndCI issues [Finishes #134209527]

Commit:
bug: fix application erro on heroku [Fixes #134209527]

Commit:
bug: fix application erro on heroku [Fixes #134209527]

Commit:
bug: fix application erro on heroku [Fixes #134209527]

Commit:
test(authentication-test): wrote test for sign in and login routes [Finishes #134209527]

Commit:
refactor(houndci): fix issues with houndci  [Finishes #134209527]

Commit:
refactor(coverage): refactor test to increase coverage [Finishes #134209527]

Commit:
refactor(coverage): refactor test to increase coverage [Finishes #134209527]

Commit:
bug(test): fix test passing without covering test [Finishes #134209527]

tomily1 commented on app/controllers/users.js:
```diff
@@ -45,47 +54,46 @@ exports.signout = function(req, res) {
+        }
+      });
   } else {
     // If user doesn't even exist, redirect to /
```
***Nitpick***
put the forward slash in a quote like '/'

tomily1 commented on public/views/play-with.html:
```diff
@@ -0,0 +1,53 @@
+<nav id="mainNav" class="navbar navbar-default navbar-fixed-top navbar-custom">
```
Good Job, nice format for html code

tomily1 commented on test/user/controller.js:
```diff
@@ -0,0 +1,66 @@
+        .expect(302, done());
+    });
+
+    it('should sign up when valid data is passed', (done) => {
```
fix this

tomily1 reviewed (dismissed):
Work on the houndCI issues and the nitpick

Commit:
feature(security): add email validation to signin input box [WIP #134209527]

Commit:
feature(security): add form validation to login and sign up forms [Finishes #134209527]

this commit also makes slight changes on the user controller and test based on review

mfalade commented on app/controllers/users.js:
```diff
@@ -139,21 +147,20 @@ exports.addDonation = function(req, res) {
+          var duplicate = false;
+          for (var i = 0; i < user.donations.length; i += 1) {
+            if (user.donations[i].crowdrise_donation_id === req.body.crowdrise_donation_id) {
+              duplicate = true;
```
For performance sakes here, you should have an early break/return statement once this condition is met as there is no need going through the remainder of the list.

So you should have something like

```javascript
if (user.donations[i].crowdrise_donation_id === req.body.crowdrise_donation_id) {
    duplicate = true;
    break; // - out of the for loop
}
```

Or better still, you could use the convenient `Array.prototype.some` method and replace this entire block
```javascript
var duplicate = false;
for (var i = 0; i < user.donations.length; i += 1) {
    if (user.donations[i].crowdrise_donation_id === req.body.crowdrise_donation_id) {
        duplicate = true;
    }
}
```
with

```javascript
const duplicate  = user.donations.some(donation => donation.crowdrise_donation_id ===  req.body.crowdrise_donation_id);
```

mfalade commented on public/views/play-with.html:
```diff
@@ -0,0 +1,53 @@
+      <i class="fa fa-bars"></i>
+      </button>
+      <a class="navbar-brand-custom" href="#">
+      <img alt="Brand" src="./img/cards-for-humanity.png" class="img-responsive" width="70">
```
Say NO to inline styling.  Move the width styling to a css file. 

Also, since you are applying a class `img-responsive` to the image, you don't want to set a static `width: 70px`. You should make this `max-width` instead.

mfalade commented on public/views/play-with.html:
```diff
@@ -0,0 +1,53 @@
+          <a href="/">HOME</a>
+        </li>
+        <li class="page-scroll">
+          <a href="/#!/#about">ABOUT THE GAME</a>
```
Why do you have a Has# before the url?

The shebang `#!` is understandable, but why the `#about` / `#charity`?

Change these to `about`/`charity`.

andela-uenabulele replied:
Thanks for your awesome comments!
`#charity` and `#about` are divs on the home page I am targeting, they do not have routes.

mfalade replied:
Ah, I see. Thanks for clarifying.

If that's the case, you don't need the shebang then.. The `href` can just be `#about`

andela-uenabulele replied:
Ok, I have tried that but it would not work on the sign in page because the div is not on that page
so I have to redirect to login page using the "/#! and then to the about div "#about"

mfalade commented on public/views/signin.html:
```diff
@@ -1,76 +1,80 @@
+      <i class="fa fa-bars"></i>
+      </button>
+      <a class="navbar-brand-custom" href="#">
+      <img alt="Brand" src="./img/cards-for-humanity.png" class="img-responsive" width="70">
```
Resist the urge to do inline styling. 

Some say it is a sign of laziness 😉

mfalade commented on public/views/signin.html:
```diff
@@ -1,76 +1,80 @@
+            <h4 class="signup-subtitle text-center">
+              Login to Play Game
+            </h4>
+            <div class="sign-error" ng-show="showError() === 'invalid'">
```
Update `showError()` to simply return a boolean.

The less logic you have in your template, the better.

mfalade commented on test/user/controller.js:
```diff
@@ -0,0 +1,65 @@
+    done();
+  });
+  describe('Login', () => {
+    it('should redirect to login page when login is click', (done) => {
```
... should redirect to login page when login is `clicked`

mfalade commented on test/user/controller.js:
```diff
@@ -0,0 +1,65 @@
+    });
+  });
+  describe('SignUp', () => {
+    it('should redirect to the sign up page when sign up is click', (done) => {
```
...when sign up is `clicked`

mfalade reviewed (changes requested):
Please address comments

Commit:
refactor(performance): refactor addDonation for better performance [WIP #134209527]

Commit:
refactor(style-guide): move inline styling to css class [WIP #134209527]

Commit:
refactor(test): fix grammar error in test statement [WIP #134209527]

Commit:
refactor: refactor login to show error message [Finishes #134209527]

Commit:
bug(houndci): fix hound ci issue [Finishes #134209527]
